### PR TITLE
Return 500 instead of 404 for general error

### LINKED ISSUE #250: must not return 404 in case of error
```
% curl -v --connect-to 'rest.fs.neo.org:443:metis2.nspcc.ru:443' https://rest.fs.neo.org/v1/objects/7yJS1QrmTsaQzVDUMjgKYnfUi7RvjCUhmnRgWL7XKn9P/by_attribute/FilePath/index.html
* Connecting to hostname: metis2.nspcc.ru
* Connecting to port: 443
* Host metis2.nspcc.ru:443 was resolved.
* IPv6: (none)
* IPv4: 149.56.16.141
*   Trying 149.56.16.141:443...
* GnuTLS ciphers: NORMAL:-ARCFOUR-128:-CTYPE-ALL:+CTYPE-X509:-VERS-SSL3.0
* found 146 certificates in /etc/ssl/certs/ca-certificates.crt
* found 440 certificates in /etc/ssl/certs
* SSL connection using TLS1.3 / ECDHE_RSA_AES_256_GCM_SHA384
*   server certificate verification OK
*   server certificate status verification SKIPPED
*   common name: rest.fs.neo.org (matched)
*   server certificate expiration date OK
*   server certificate activation date OK
*   certificate public key: RSA
*   certificate version: #3
*   subject: CN=rest.fs.neo.org
*   start date: Thu, 22 Aug 2024 22:17:54 GMT
*   expire date: Wed, 20 Nov 2024 22:17:53 GMT
*   issuer: C=US,O=Let's Encrypt,CN=R11
* ALPN: server accepted h2
* Connected to metis2.nspcc.ru (149.56.16.141) port 443
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://rest.fs.neo.org/v1/objects/7yJS1QrmTsaQzVDUMjgKYnfUi7RvjCUhmnRgWL7XKn9P/by_attribute/FilePath/index.html
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: rest.fs.neo.org]
* [HTTP/2] [1] [:path: /v1/objects/7yJS1QrmTsaQzVDUMjgKYnfUi7RvjCUhmnRgWL7XKn9P/by_attribute/FilePath/index.html]
* [HTTP/2] [1] [user-agent: curl/8.10.1]
* [HTTP/2] [1] [accept: */*]
> GET /v1/objects/7yJS1QrmTsaQzVDUMjgKYnfUi7RvjCUhmnRgWL7XKn9P/by_attribute/FilePath/index.html HTTP/2
> Host: rest.fs.neo.org
> User-Agent: curl/8.10.1
> Accept: */*
> 
* Request completely sent off
< HTTP/2 404 
< server: nginx
< date: Mon, 21 Oct 2024 15:37:17 GMT
< content-type: application/json
< content-length: 135
< strict-transport-security: max-age=63072000; includeSubDomains
< x-content-type-options: nosniff
< x-frame-options: SAMEORIGIN
< access-control-allow-origin: *
< 
{"code":1024,"message":"read object list failed: status: code = 1024 message = connection to the RPC node has been lost","type":"API"}
* Connection #0 to host metis2.nspcc.ru left intact
```

```
Oct 21 15:37:17 metis2 neofs-rest-gw[701]: error        handlers/api.go:128        read object list failed        {"error": "status: code = 1024 message = connection to the RPC node has been lost"}
Oct 21 15:37:17 metis2 neofs-rest-gw[701]: error        handlers/newObjects.go:235        failed to close resource        {"error": "status: code = 1024 message = connection to the RPC node has been lost"}
```

### PR DESCRIPTION
**Pull request by smallhive**
Closes #250.

### CODE REVIEW

File: handlers/newObjects.go [hunk 1]
```diff
@@ -227,7 +227,7 @@ func (a *RestAPI) NewGetByAttribute(ctx echo.Context, containerID apiserver.Cont
 	res, err := a.search(ctx.Request().Context(), btoken, cnrID, attrKey, attrVal, object.MatchStringEqual)
 	if err != nil {
 		resp := a.logAndGetErrorResponse("could not search for objects", err)
-		return ctx.JSON(http.StatusNotFound, resp)
+		return ctx.JSON(http.StatusInternalServerError, resp)
```
Comment (roman-khimov) [inline handlers/newObjects.go:230]: Errors can be different. 500 is also different from 502.

Comment (smallhive) [reply handlers/newObjects.go:230]: Does it make sense to divide 500 and 502? If we do, should I check the error message `connection to the RPC node has been lost` to return 502?

Comment (roman-khimov) [reply handlers/newObjects.go:230]: NeoFS error code is sufficient. Keep 500 for local (gateway) problems and return 502 if you have some strange NeoFS error returned.

File: handlers/newObjects.go [hunk 2]
```diff
@@ -227,7 +227,11 @@ func (a *RestAPI) NewGetByAttribute(ctx echo.Context, containerID apiserver.Cont
 	res, err := a.search(ctx.Request().Context(), btoken, cnrID, attrKey, attrVal, object.MatchStringEqual)
 	if err != nil {
 		resp := a.logAndGetErrorResponse("could not search for objects", err)
-		return ctx.JSON(http.StatusNotFound, resp)
+		if strings.Contains(err.Error(), "code = 1024") {
```
Comment (roman-khimov) [inline handlers/newObjects.go:230]: `strings.Contains` is banned for this. You have a whole https://pkg.go.dev/github.com/nspcc-dev/neofs-sdk-go@v1.0.0-rc.12/client/status for this. Plus https://github.com/nspcc-dev/neofs-api/blob/master/status/types.proto.

File: handlers/newObjects.go [hunk 3]
```diff
@@ -227,7 +227,11 @@ func (a *RestAPI) NewGetByAttribute(ctx echo.Context, containerID apiserver.Cont
...
+			return ctx.JSON(http.StatusBadGateway, resp)
+		}
+
+		return ctx.JSON(http.StatusInternalServerError, resp)
```
Comment (roman-khimov) [inline handlers/newObjects.go:234]: Suppose it's 2048. Or 1027. Or 3072. Different things, right?

File: handlers/newObjects.go (hunk 2 above)
Comment (smallhive) [reply handlers/newObjects.go:230]: Replaced with conversion by status

Comment (roman-khimov) [reply handlers/newObjects.go:230]: Needs to be updated.

Comment (smallhive) [reply handlers/newObjects.go]: Rebased and updated

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**